src/cider/nrepl/middleware/debug.clj

```diff
@@ -346,20 +350,40 @@
 (defmacro breakpoint
   "Send the result of form and its coordinates to the client.
   Sends a response to the message stored in debugger-message."
+  [form {:keys [coor]} original-form]
+  `(let [val# ~form]
+     (cond
+       (skip-breaks? ~coor) val#
+       ;; The length of `coor` is a good indicator of current code
+       ;; depth.
+       (= (first @*skip-breaks*) :trace)
+       (do (print-step-indented ~(count coor) '~original-form val#)
+           val#)
+       ;; Nothing special. Here's the actual breakpoint logic.
+       :else (->> (pr-short val#)
+                  (assoc *extras* :debug-value)
+                  (read-debug-command val#)))))
+
+(defmacro maybe-breakpoint
+  "Same as `breakpoint` unless there is a condition attached to the
+  form as `:break/when` metadata, and it is falsy."
   [form {:keys [coor] :as extras} original-form]
-  `(with-debug-bindings ~extras
-     (let [val# ~form]
-       (cond
-         (skip-breaks? ~coor) val#
-         ;; The length of `coor` is a good indicator of current code
-         ;; depth.
-         (= (first @*skip-breaks*) :trace)
-         (do (print-step-indented ~(count coor) '~original-form val#)
-             val#)
-         ;; Nothing special. Here's the actual breakpoint logic.
-         :else (->> (pr-short val#)
-                    (assoc *extras* :debug-value)
-                    (read-debug-command val#))))))
+  (let [has-condition? (contains? (meta form) :break/when)
+        condition      (:break/when (meta form))
+        parent-coor    (vec (butlast coor))]
+    (if has-condition?
+      `(with-debug-bindings ~extras
+         (let [skip?# (not ~condition)]
+           ;; If there is a condition and it is falsy, we need to skip
+           ;; the current level (:deeper than parent coor), but only
+           ;; once. Next time, we need to test the condition again.
+           (binding [*skip-breaks* (if skip?#
```

Comment:
Do we need :deeper? Isn't that the same as skipping everything inside this `binding`?